### PR TITLE
Add WatchedFileHandler

### PR DESCRIPTION
**Pull request by aminalaee**
Adds `WatchedFileHandler` for compatibility with logging.

As per the docs https://docs.python.org/3/library/logging.handlers.html#watchedfilehandler I think the WatchedFileHandler and other new handlers following it should go in 'picologging.handlers' to be compatible with the logging module.

### CODE REVIEW

File: tests/test_filehandler.py [hunk 1]
```diff
@@ -11,3 +11,16 @@ def test_filehandler(tmp_path):
...
+
+
+def test_watchedfilehandler(tmp_path):
+    log_file = tmp_path / 'log.txt'
+    handler = picologging.WatchedFileHandler(log_file)
+    logger = picologging.getLogger('test')
+    logger.setLevel(picologging.DEBUG)
+    logger.addHandler(handler)
+    logger.warning('test')
+    handler.close()
```
Comment (aminalaee) [inline tests/test_filehandler.py:23]: The test needs to be updated.

**Comment by tonybaloney** [comment]
hey thanks for the contribution! I'll review this early this week. I see you've raised 3 PRs, should I review them in any particular order?

**Comment by tonybaloney** [comment]
Please take a look at the test failures on Windows that the CI is reporting

**Comment by aminalaee** [comment]
@tonybaloney No I think any order will do, they will need rebasing but it should be fine.

File: src/picologging/handlers.py [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+
+    def emit(self, record):
+        """
+        Emit a record.
+
+        If underlying file has changed, reopen the file before emitting the
+        record to it.
+        """
+        self.reopenIfNeeded()
+        picologging.FileHandler.emit(self, record)
```
Comment (tonybaloney) [inline src/picologging/handlers.py:62]: Please change to 

```python
super().emit(record)
```

**Review by tonybaloney — CHANGES_REQUESTED** [review]

File: src/picologging/handlers.py (hunk 2 above)
Comment (aminalaee) [reply src/picologging/handlers.py]: Done.

**Review by tonybaloney — APPROVED** [review]

**Comment by tonybaloney** [comment]
Brilliant! Thanks for the PR.

GitHub also tells me this is your first merged PR in the Microsoft org so thanks on behalf of Microsoft too

**Comment by aminalaee** [comment]
Thanks for the amazing work 🚀

**Pull request merged**